plugins/woocommerce/src/Internal/Admin/Homescreen.php

```diff
@@ -50,6 +51,82 @@ public function __construct() {
 			add_action( 'admin_head', array( $this, 'update_link_structure' ), 20 );
 		}
 		add_filter( 'woocommerce_admin_preload_options', array( $this, 'preload_options' ) );
+		add_filter( 'woocommerce_admin_shared_settings', array( $this, 'maybe_set_default_shipping_options_on_home' ), 9999 );
+	}
+
+	/**
+	 * Set free shipping in the same country as the store default
+	 * Flag rate in all other countries when any of the following conditions are ture
+	 *
+	 * - The store sells physical products, has JP and WCS installed and connected, and is located in the US.
+	 * - The store sells physical products, and is not located in US/Canada/Australia/UK (irrelevant if JP is installed or not).
+	 * - The store sells physical products and is located in US, but JP and WCS are not installed.
+	 *
+	 * @param array $settings shared admin settings.
+	 * @return array
+	 */
+	public function maybe_set_default_shipping_options_on_home( $settings ) {
+		$current_screen = get_current_screen();
+
+		// Abort if it's not the homescreen.
+		if ( ! isset( $current_screen->id ) || 'woocommerce_page_wc-admin' !== $current_screen->id ) {
+			return $settings;
+		}
+
+		// Abort if we already created the shipping options.
+		$already_created = get_option( 'woocommerce_admin_created_default_shipping_zones' );
+		if ( 'yes' === $already_created ) {
+			return $settings;
+		}
+
+		if ( false === in_array( 'physical', $settings['onboarding']['profile']['product_types'] ?? array(), true ) ) {
+			return $settings;
+		}
+
+		$country_code = explode( ':', $settings['preloadSettings']['general']['woocommerce_default_country'] ?? '' )[0];
```

Comment:
This is a small thing. We could just use `wc_format_country_state_string` function here.

```suggestion
		$country_code = wc_format_country_state_string( $settings['preloadSettings']['general']['woocommerce_default_country'] )['country'];
```

Reply:
Nice! Thank you 👍